File: лаба2/Swift/laba2_1zadanie.swift
import Foundation

// Проверка, является ли число простым
func isprostoe(_ n: Int) -> Bool {
    for i in 2...Int(sqrt(Double(N))) {
        if n % i == 0 {
            return false  // Делится на i без остатка - не простое
        }
    }
    return n > 1  // Числа <= 1 не считаются простыми
}

// Поиск ближайшего простого числа к n
func near(_ n: Int) -> Int {
    if isprostoe(N) {
        return n  // Если n простое - возвращаем его
    }
    var lower = n - 1
    var upper = n + 1

    while true {
        if isprostoe(lower) {
            return lower  // Нашли простое число ниже n
        }
        if isprostoe(upper) {
            return upper  // Нашли простое число выше n
        }
        lower -= 1
        upper += 1
    }
}

func main() {
    print("Пример: ")
    guard let input = readLine(), let n = Int(input) else {
        print("Неправильный формат ввода")
        return
    }
    let num = near(n)          // Ищем ближайшее простое
    let res = n - num          // Разница

    print("Ближайшее простое: \(num)\n")
    print("Результат: \(res)\n")
}

main()

File: лаба2/Rust/laba2_1zadanie.rs
use std::io;

// Проверка, является ли число простым
fn isprostoe(n: i32) -> bool {
    // Перебираем делители от 2 до квадратного корня из n
    for i in 2..=((n as f64).sqrt() as i32) {
        if n % i == 0 {
            return false;  // Если делится без остатка - не простое
        }
    }
    n > 1  // Числа меньше или равные 1 не простые
}

// Поиск ближайшего простого числа к n
fn near(n: i32) -> i32 {
    if isprostoe(n) {
        return n;  // Если n простое - возвращаем его
    }
    let mut lower = n - 1;
    let mut upper = n + 1;

    loop {
        if isprostoe(lower) {
            return lower;  // Нашли простое число ниже n
        }
        if isprostoe(upper) {
            return upper;  // Нашли простое число выше n
        }
        lower -= 1;
        upper += 1;
    }
}

fn main() {
    println!("Пример: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: i32 = input.trim().parse().expect("Please type a number!");

    let num = near(n);           // Ищем ближайшее простое число
    let res = n - num;           // Вычисляем разницу

    println!("Ближайшее простое: {}", num);
    println!("Результат: {}", res);
}

